string tpName
int64 tpValue

